CMakeLists.txt

```diff
@@ -581,8 +580,9 @@ if (WIN32)
 	)
 
 	target_compile_definitions(torrent-rasterbar
-		PUBLIC
+		PRIVATE
```

Comment:
I know little about Windows includes, but it seems strange to me that these two control definitions have different visibility.

Reply:
This define determines which version of the windows runtime to compile against. If this is public, it forces all clients of libtorrent to also target vista, which I don't think is reasonable. IIUC, private means it only affects the libtorrent build itself

Reply:
But `WIN32_LEAN_AND_MEAN` casts restrictions on the API too. Why these two macros differ in scopes? My point is purely about their conceptual similarity (both cancel some declaration out from Windows include files).

Reply:
that's true. However, `WIN32_LEAN_AND_MEAN` is (likely) required to not break *inluding* a libtorrent header. I'm not sure but I seem to recall that there are boost.asio headers that in turn include windows.h. It's also less controversial and restrictive, since (my impression at least) is that everyone building on windows has defined this for decades to get a slightly saner build environment. So it could be considered a service to clients to provide this for them :)

Reply:
Thanks for the explanations!